libretroshare/src/pgp/pgphandler.cc

```diff
@@ -1978,7 +1978,7 @@ bool PGPHandler::removeKeysFromPGPKeyring(const std::set<RsPgpId>& keys_to_remov
 	char template_name[_pubring_path.length()+8] ;
 	sprintf(template_name,"%s.XXXXXX",_pubring_path.c_str()) ;
 	
-	if(mktemp(template_name) == NULL)
+    if(mkstemp(template_name) == -1)
```

Comment:
this creates a memory leak since mkstemp "returns an open file descriptor for the file" that should be closed  ( http://man7.org/linux/man-pages/man3/mkstemp.3.html )
also this might not work on windows. I had a patch set for this but i never got merged